### PR TITLE
Vendor SDK for function-logs

### PR DESCRIPTION
**Pull request by martindekov**
Vendor faas-cli SDK for function logs
instead of creating custom http requests.
Also rewriting unit tests and removing
excess error handling which is moved in
the SDK and is excessive as we don't
parse JSON into the function anymore

Signed-off-by: Martin Dekov <mvdekov@gmail.com>

## Description

Part of #609 

## How Has This Been Tested?

On my own private instance with image containing the change `martindekov/function-logs:0.3.0-rc3`:

`functions/function-logs:0.0.2` commit log output:

https://github.com/martindekov/push2/runs/509201597
image

`martindekov/function-logs:0.3.0-rc3` commit log output:

https://github.com/martindekov/push2/runs/509210079
image

UI's `Build Logs` with same image:

## How are existing users impacted? What migration steps/scripts do we need?

N/A Internal change

## Checklist:

I have:

- [ ] updated the documentation and/or roadmap (if required)
- [x] read the [CONTRIBUTION](https://github.com/openfaas/faas/blob/master/CONTRIBUTING.md) guide
- [x] signed-off my commits with `git commit -s`
- [ ] added unit tests
- [x] augmented unit tests

### CODE REVIEW

**Comment by alexellis** [comment]
Just a comment on the wrapping of the commit message. What width are you using as a template? It seems too narrow.

**Comment by alexellis** [comment]
@martindekov FYI I think the testing of the build logs may be irrelevant here since they are not using the OpenFaaS REST API, they are simply the output of the of-builder service. Can you confirm if that's your understanding too?

File: function-logs/handler.go [hunk 1]
```diff
@@ -1,20 +1,34 @@
...
 )
 
+type FaaSAuth struct{}
+
+func (auth *FaaSAuth) Set(req *http.Request) error {
+	return sdk.AddBasicAuth(req)
+}
+
+var (
+	timeout   = 3 * time.Second
```
Comment (alexellis) [inline function-logs/handler.go]: This may not be enough for when the system is under contention, perhaps 5s might be a safer bet?

**Comment by martindekov** [comment]
I will need to re-visit this change as per your comment.

**Comment by alexellis** [comment]
Thanks Martin

File: function-logs/handler.go (hunk 1 above)
Comment (martindekov) [reply function-logs/handler.go]: Increased to 5 :+1:

File: function-logs/handler.go [hunk 2]
```diff
@@ -40,136 +54,86 @@ func Handle(req []byte) string {
...
 	}
 
 	if !allowed {
 		log.Fatalf("requested function %q could not be found or you are not allowed to access it", function)
 	}
 
-	formattedLogs, fmtErr := getFormattedLogs(gatewayURL, function)
+	client := faasSDK.NewClient(&FaaSAuth{}, gatewayURL, nil, &timeout)
+
+	formattedLogs, fmtErr := getFormattedLogs(*client, function)
```
[the remark this answers is not in the document]
Comment (martindekov) [reply function-logs/handler.go]: Yeah Vivek will add this :+1:

[the remark this answers is not in the document]
Comment (martindekov) [reply function-logs/handler.go]: Done :+1:

**Pull request merged**